src/daft-core/src/utils/arrow.rs

```diff
@@ -95,15 +109,33 @@ pub fn cast_array_for_daft_if_needed(
     arrow_array: Box<dyn arrow2::array::Array>,
 ) -> Box<dyn arrow2::array::Array> {
     match coerce_to_daft_compatible_type(arrow_array.data_type()) {
-        Some(coerced_dtype) => cast::cast(
-            arrow_array.as_ref(),
-            &coerced_dtype,
-            cast::CastOptions {
-                wrapped: true,
-                partial: false,
-            },
-        )
-        .unwrap(),
+        Some(coerced_dtype) => match coerced_dtype {
+            // TODO: Consolidate Map to use the same cast::cast method as other datatypes.
+            // Currently, Map is not supported in Arrow2::compute::cast, so this workaround is necessary.
+            // A known limitation of this workaround is that it does not handle nested maps.
```

Comment:
Reading through the code, this seems to handle nested maps because it calls `cast_array_for_daft_if_needed` again on its child `.field`?

Reply:
in arrow2 the child `.field` of a maparray is a structarray, so calling `cast_array_for_daft_if_needed` on it will branch to the non-map path of using `cast::cast`, and if inside of that structarray is a maparray, then it will fail inside `cast::cast`